tests/testTableArchives.cc

```diff
@@ -642,3 +644,31 @@ BOOST_AUTO_TEST_CASE(ArchiveImporter) {
     lsst::afw::image::Exposure<float> exposure("tests/data/archiveImportTest.fits");
     BOOST_CHECK(!exposure.getPsf());
 }
+
+
+BOOST_AUTO_TEST_CASE(ArchiveMetadata) {
+    // Test that we only write one AR_NAME header key for each class saved in an HDU, not each instance
+    // (see DM-7221).
+    using namespace lsst::afw::table::io;
+    using namespace lsst::afw::fits;
+    using namespace lsst::afw::detection;
+    using namespace lsst::afw::geom;
+    OutputArchive outArchive;
+    outArchive.put(std::make_shared<Footprint>(Box2I(Point2I(2, 3), Point2I(5, 4))));
+    outArchive.put(std::make_shared<Footprint>(Box2I(Point2I(1, 2), Point2I(7, 6))));
+    outArchive.put(std::make_shared<HeavyFootprint<float>>(Footprint(Box2I(Point2I(1, 2), Point2I(7, 6)))));
+    MemFileManager manager;
+    Fits outFits(manager, "w", Fits::AUTO_CHECK);
+    outArchive.writeFits(outFits);
+    outFits.closeFile();
+    Fits inFits(manager, "r", Fits::AUTO_CHECK);
+    inFits.setHdu(3);
+    lsst::daf::base::PropertyList metadata;
+    inFits.readMetadata(metadata);
+    inFits.closeFile();
+    auto names = metadata.getArray<std::string>("AR_NAME");
```

Comment:
I'll add the comment.  I'm pretty certain this test would have failed before the fix, but I didn't actually do it in that order.